### DISCUSSION TITLE
How can we get inbound shipping fees for a shipment?

### DISCUSSION THREAD

**Discussion by ProNotion** [amzn/selling-partner-api-models#4411] [Q&A]
It feels like it is being made impossibly difficult to obtain specific inbound shipping data with the available APIs. My latest challenge is trying to get fees for a shipment. Fees are returned in the Finances API but without any reference to the shipment in which they relate. How can we tie the fees to the shipment to which they relate?

**Comment by kevnhowe** [comment]
I have been looking for this data for more than a year. It boggles my mind that we don't have this information. I cannot effectively audit my payment settlement charges against Amazon's reported per unit "Inbound Shipping Fees." This also makes it impossible to accurately calculate my GPM per ASIN and have to trust whatever number Amazon is reporting on the SKU economics report will agree with our actual inbound shipping charges.

**Comment by ProNotion** [comment]
@kevnhowe It does feel like it is intentional because of how significant an oversight some of it is. I have another ongoing problem to do with inbound shipment dates too and can see no logical reason for withholding them. Incredibly frustrating!

**Comment by puppsupr** [comment]
HI @ProNotion,
Are you looking for the carrier fees to inbound ?

If yes, you can call the below APIs,

1. [getInboundPlan](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#getinboundplan)
2. To get quote details - [listTransportationOptions](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-reference#listtransportationoptions)

Also using the new Finances v2024-06-19, is [SHIPMENT_ID](https://developer-docs.amazon.com/sp-api/docs/finances-api-v2024-06-19-reference#relatedidentifiername) not useful ?

**Reply by kevnhowe** [reply]
Yes. Shipment Id does get us from a financial event to a shipment. This is what I've been looking for. :)

**Comment by puppsupr** [comment]
Great ! Please let us know if you need further help.
